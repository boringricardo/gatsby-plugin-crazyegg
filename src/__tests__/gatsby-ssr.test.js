const { onRenderBody } = require('../gatsby-ssr');

describe('Adds <script> for site to head', () => {
  beforeEach(() => {
    process.env = { NODE_ENV: 'production' };
  });

  it('adds the script to head', async () => {
    const pluginOptions = {
      accountNumber: '123',
    };
    const setHeadComponents = jest.fn();

    await onRenderBody({ setHeadComponents }, pluginOptions);

    expect(setHeadComponents).toMatchSnapshot();
    expect(setHeadComponents).toHaveBeenCalledTimes(1);
  });

  it('does not add <script> if the account number is empty', async () => {
    const pluginOptions = {};
    const setHeadComponents = jest.fn();

    await onRenderBody({ setHeadComponents }, pluginOptions);

    expect(setHeadComponents).toMatchSnapshot();
    expect(setHeadComponents).toHaveBeenCalledTimes(0);
  });

  it('correctly sets the account number on the script', async () => {
    const pluginOptions = {
      accountNumber: '123',
    };

    const head = await onRenderBody({ setHeadComponents }, pluginOptions);
    console.log(head);
  });

  it(`doesn't add the script if NODE_ENV is not "production"`, async () => {
    const pluginOptions = {
      accountNumber: '123',
    };
    const setHeadComponents = jest.fn();

    await onRenderBody({ setHeadComponents }, pluginOptions);

    expect(setHeadComponents).toMatchSnapshot();
    expect(setHeadComponents).toHaveBeenCalledTimes(0);
  });
})