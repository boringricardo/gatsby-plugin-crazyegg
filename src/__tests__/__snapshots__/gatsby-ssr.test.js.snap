// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds <script> for site to head adds the script to head 1`] = `[MockFunction]`;

exports[`Adds <script> for site to head does not add <script> if the account number is empty 1`] = `[MockFunction]`;

exports[`Adds <script> for site to head doesn't add the script if NODE_ENV is not "production" 1`] = `[MockFunction]`;
